### DISCUSSION TITLE
How to create an Action to have custom logic in addition to using the HTTP-Processor class ?

### DISCUSSION THREAD

**Discussion by marifrahman** [apioo/fusio#547] [Q&A]
Hi @chriskapp,
I want to create an Action having the HTTP-Processor class.
I have a POST operation that passThrough the request body to api via my Action. 
I want to manipulate this request body - what user passes to operation  before passing this to Action - before calling the external api.
What's the best way to do it ?

**Comment by chriskapp** [comment]
Hi @marifrahman you could try to to use the [Util-Json-Patch](https://docs.fusio-project.org/docs/backend/api/action/util-json-patch) action which then invokes your HTTP processor action, through this action you can modify the JSON payload using [JSON Patch](https://jsonpatch.com/). As alternative you can of course always create a custom action where you can modify the complete payload.
